app/src/features/surveys/view/survey-animals/TelemetryDeviceForm.tsx

```diff
@@ -0,0 +1,104 @@
+import { FormHelperText } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import CustomTextField from 'components/fields/CustomTextField';
+import SingleDateField from 'components/fields/SingleDateField';
+import TelemetrySelectField from 'components/fields/TelemetrySelectField';
+import { Form, useFormikContext } from 'formik';
+import useDataLoader from 'hooks/useDataLoader';
+import { useTelemetryApi } from 'hooks/useTelemetryApi';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import { IAnimalTelemetryDevice } from './animal';
+
+const TelemetryDeviceForm = () => {
+  const { values, setStatus } = useFormikContext<IAnimalTelemetryDevice>();
+  const [bctwErrors, setBctwErrors] = useState<Record<string, string | undefined>>({});
+  const api = useTelemetryApi();
+  const {
+    data: deviceData,
+    load: loadDevice,
+    refresh
+  } = useDataLoader(() => api.devices.getDeviceDetails(values.device_id));
+
+  useEffect(() => {
+    loadDevice();
+    refresh();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [values.device_id]);
+
+  useEffect(() => {
+    const errors: { device_make?: string; attachment_start?: string } = {};
+    if (deviceData?.device && deviceData.device?.device_make !== values.device_make) {
+      errors.device_make = `Submitting this form would change the registered device make of device ${values.device_id}, which is disallowed.`;
+    }
+
+    const existingDeployment = deviceData?.deployments?.find(
+      (a) =>
+        (moment(values.attachment_start).isSameOrAfter(moment(a.attachment_start)) &&
```

Comment:
In the future, avoid `moment` (**when possible!**) as it is a deprecated library. Currently I don't think we have an alternative library installed; Just pointing this out.

Reply:
Agreed that it would be better to use a different library but I believe this usage is necessary here.